indexdata.go

```diff
@@ -184,26 +188,32 @@ func (d *indexData) calculateStatsForRepoIndex(i int) RepoListEntry {
 		OtherBranchesNewLinesCount: otherCount,
 	}
 	return RepoListEntry{
-		Repository:    d.repoMetaData[i],
+		Repository:    d.repoMetaData[d.repos[start]],
 		IndexMetadata: d.metaData,
 		Stats:         stats,
 	}
 }
 
 func (d *indexData) calculateStats() {
 	d.repoListEntry = make([]RepoListEntry, 0, len(d.repoMetaData))
+	var start, end uint32
 	for i := 0; i < len(d.repoMetaData); i++ {
-		d.repoListEntry = append(d.repoListEntry, d.calculateStatsForRepoIndex(i))
+		// determine the file range for repo i
+		for end < uint32(len(d.repos)) && d.repos[end] == uint16(i) {
+			end++
+		}
+		d.repoListEntry = append(d.repoListEntry, d.calculateStatsForFileRange(start, end))
+		start = end
```

Comment:
your baking in the invariant htat data will appear in the same order as repoListEntries. I don't think we have this assumption anywhere else?

Reply:
I added an explicit cxheck for this and returned an error if we don't